File: Editor/src/main/java/org/turbomedia/turboedit/editor/events/ShortcutHandler.java
package org.turbomedia.turboedit.editor.events;

import javafx.scene.Scene;
import javafx.scene.input.KeyCode;
import javafx.scene.input.KeyEvent;
import org.turbomedia.turboedit.editor.windows.PreferencesWindow;
import org.turbomedia.turboedit.editor.windows.RenderQueueWindow;

public class ShortcutHandler {
    public static void RegisterGlobalShortcuts(Scene scene) {
        scene.addEventFilter(KeyEvent.KEY_PRESSED, (event) -> {
            if (event.isControlDown()) {
                if (event.getCode() == KeyCode.Q) {
                    new RenderQueueWindow();
                } else if (event.getCode() == KeyCode.P) {
                    new PreferencesWindow();
                }
            } else {
                if (event.getCode() == KeyCode.SPACE) {
                    EventSystem.CallEvent(EventType.PLAYBACK_TOGGLE, null);
                }
            }
        });
    }
}

File: Editor/src/main/kotlin/turboedit/editor/events/ShortcutHandler.kt
package turboedit.editor.events

import javafx.event.EventHandler
import javafx.scene.Scene
import javafx.scene.input.KeyCode
import javafx.scene.input.KeyEvent
import turboedit.editor.windows.RenderQueueWindow
import turboedit.editor.windows.preferences.PreferencesWindow

object ShortcutHandler {
    fun registerGlobalShortcuts(scene: Scene) {
        scene.addEventFilter<KeyEvent?>(KeyEvent.KEY_PRESSED, EventHandler { event: KeyEvent ->
            if (event.isControlDown) {
                if (event.code == KeyCode.Q) {
                    RenderQueueWindow()
                } else if (event.code == KeyCode.P) {
                    PreferencesWindow()
                }
            } else {
                if (event.code == KeyCode.SPACE) {
                    EventSystem.callEvent(EventType.PLAYBACK_TOGGLE)
                }
            }
        })
    }
}
